app/validators/unique_cross_model_slug_validator.rb

```diff
@@ -0,0 +1,45 @@
+##
+# Validates if the give attribute is used across the reserved spaces.
+class UniqueCrossModelSlugValidator < ActiveModel::EachValidator
+  class_attribute :model_and_attribute_name_for_uniquness_test
+
+  # Why a class attribute?  Allow for other implementations to extend
```

Comment:
Could you elaborate on the possible extensions you have in mind and why a regular Ruby method that subclasses can override wasn't an option for that? It seems a bit unlikely to me that someone would make use of the `UniqueCrossModelSlugValidator.model_and_attribute_name_for_uniqueness_test = ... ` accessor `class_attribute` provides as people don't usually interact with validators so directly.

Reply:
I think I got ahead of myself in preliminary refactoring (this started as a constant).  But I'll go into my thoughts.

I'm coming from a past community that released their application as a Rails Engine.  You would basically mount the engine at "/", so the code was not visible in the application.  This meant we needed to expose more configuration options and eschewed the idea of overriding classes.  We tried to refactor/build that engine with interfaces in mind.

I'm uncertain how others are forking the application and making their modifications (and keeping up to date with our modifications).

The class attribute allows an implementor to add a line to their application initializer to append a Model / Attribute pair to the existing system (see below):

```ruby
UniqueCrossModelSlugValidator.model_and_attribute_name_for_uniqueness_test[NewModel] = :slug
```

Were they to overwrite the method (by going into the validator) then they would have potential future merge conflicts to address.

I suppose this is my baby-step into the larger architectural question of "How do we envision folks extending the Forem application?"

Reply:
Thanks for the context, @jeremyf. Personally, I would prefer if we make decisions that go in the direction of overall app architecture in a more collaborative way, i.e. via a forem.team discussion. In the meantime, a constant wrapped in a method would be more in line with what we have elsewhere.

Reply:
I made the change, and definitely will go the route of discussions.

Reply:
Sorry, when you said "made the change" I thought you changed the code back to using a constant. Apparently you didn't, so I'm not quite sure what you actually meant/if I misinterpreted your statement?